backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/InventoryService.java

```diff
@@ -72,13 +77,18 @@ public class InventoryService
 	@Getter
 	private final InventoryRepository inventoryRepository;
 	private final SourceHUsService sourceHUsService;
+	private final ModularContractProvider modularContractProvider;
+
 
 	private static final AdMessageKey MSG_EXISTING_LINES_WITH_DIFFERENT_HU_AGGREGATION_TYPE = AdMessageKey.of("de.metas.handlingunits.inventory.ExistingLinesWithDifferentHUAggregationType");
 
-	public InventoryService(@NonNull final InventoryRepository inventoryRepository, @NonNull final SourceHUsService sourceHUsService)
+	public InventoryService(@NonNull final InventoryRepository inventoryRepository,
+			@NonNull final SourceHUsService sourceHUsService,
+			final ModularContractProvider modularContractProvider)
```

Comment:
hint: you might what to annotate all 3 service fields with nonnull and then use required args constructor annotation

Reply:
done